src/Symfony/Bridge/Doctrine/Security/RememberMe/DoctrineTokenProvider.php

```diff
@@ -132,10 +133,10 @@ public function createNewToken(PersistentTokenInterface $token)
             'lastUsed' => $token->getLastUsed(),
         ];
         $paramTypes = [
-            'class' => \PDO::PARAM_STR,
-            'username' => \PDO::PARAM_STR,
-            'series' => \PDO::PARAM_STR,
-            'value' => \PDO::PARAM_STR,
+            'class' => class_exists(ParameterType::class) ? ParameterType::STRING : \PDO::PARAM_STR,
```

Comment:
ParameterType is part of DBAL 2.7+. and DBAL 2.6 is EOLed since years (latest 2.x is 2.11 IIRC). I suggest bumping the min DBAL version to 2.7 instead of 2.6 to avoid such check.

Reply:
> latest 2.x is 2.11 IIRC

2.13 🤓 

> I suggest bumping the min DBAL version to 2.7 instead of 2.6 to avoid such check.

Fair enough. That would only solve it for this class though. In the cache component, we might encounter a real PDO instance, so we still need the ternaries there.

Reply:
but we can still simplify the condition with just a `instanceof` to distinguish DBAL and PDO.

Reply:
That… is correct. 👍🏻